internal/providers/msentraid/msentraid.go

```diff
@@ -143,30 +143,13 @@ func (p Provider) getGroups(token *oauth2.Token) ([]info.Group, error) {
 
 	// Get the groups (only the groups, not directory roles or administrative units, because that would require
 	// additional permissions) which the user is a member of.
-	graphGroups, err := client.Me().TransitiveMemberOf().GraphGroup().Get(context.Background(), nil)
+	graphGroups, err := getAllUserGroups(client)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get user groups: %v", err)
 	}
 
 	var groups []info.Group
-	for _, obj := range graphGroups.GetValue() {
-		unknown := "Unknown"
-		msGroup, ok := obj.(*msgraphmodels.Group)
-		if !ok {
-			id, oType := obj.GetId(), obj.GetOdataType()
-			if id == nil {
-				id = &unknown
-			}
-			if oType == nil {
-				oType = &unknown
-			}
-			slog.Debug(fmt.Sprintf(
-				"Found non-group object with ID: %q of type: %q in graphsdk response. Ignoring it",
-				*id, *oType,
-			))
-			continue
-		}
```

Comment:
Note that we got rid of this check by using the `msgraphmodels.Groupable` interface (which is the type of the `ItemTransitiveMemberOfGraphGroupRequestBuilder.Get` return value) instead of `*msgraphmodels.Group`.